src/systems/filecoin_markets/storage_market/storage_market_actor.go

```diff
@@ -312,6 +317,44 @@ func (a *StorageMarketActorCode_I) GetLastDealExpirationFromDealIDs(rt Runtime,
 	return lastDealExpiration
 }
 
+func (a *StorageMarketActorCode_I) GetUnsealedCIDForDealIDs(rt Runtime, sectorSize util.UVarint, dealIDs []deal.DealID) vmr.InvocOutput {
+	pieceInfos := make([]*sector.PieceInfo_I, len(dealIDs))
+
+	h, st := a.State(rt)
+
+	var totalPieceSize util.UInt
+
+	for index, deal := range st.Deals() {
+		proposal := deal.Proposal()
+		pieceSize := util.UInt(proposal.PieceSize().Total())
+		totalPieceSize += pieceSize
+		pieceInfos[index].PieceCID_ = proposal.PieceCID()
+		pieceInfos[index].Size_ = pieceSize
+	}
+
+	Release(rt, h, st)
+
+	maxAllowablePadding := sectorSize / MAX_ALLOWABLE_PADDING_DENOMINATOR
+	minTotalPiece := sectorSize - maxAllowablePadding
+
+	if totalPieceSize < minTotalPiece {
+		return rt.ErrorReturn(exitcode.InvalidSectorPacking)
+	}
+
+	SPS := new(proving.StorageProvingSubsystem_I)
+	ret := SPS.ComputeUnsealedSectorCID(sectorSize, pieceInfos)
+
+	switch ret.Which() {
+	case proving.StorageProvingSubsystem_ComputeUnsealedSectorCID_FunRet_Case_unsealedSectorCID:
+		return rt.ValueReturn(util.Bytes(sector.UnsealedSectorCID(ret.As_unsealedSectorCID())))
+	case proving.StorageProvingSubsystem_ComputeUnsealedSectorCID_FunRet_Case_err:
+		return rt.ErrorReturn(exitcode.InvalidSectorPacking)
+	}
+
+	// Unreachable -- just make the compiler happy.
```

Comment:
adding the `default` case to the switch is probably more idiomatic

Reply:
Unfortunately, that also does not seem to save me from `missing return at end of function`.